### PR TITLE
Add conv1d to burn-import (onnx)

### PR DESCRIPTION
**Pull request by mrvollger**
### Checklist

- [ x ] Confirm that `run-checks.sh` has been executed.
I have run this and nearly all tests pass except for ones associated with `burn_wgpu`. However I dont think this has anything to do with the new code and just my system. To confirm this I checked and the same tests fail if I switch to the main branch without my updates, so I think it is okay.

### Changes

I added a Conv1d as a new operation to burn-import. I largely followed  Conv2d as an example and followed the instructions in the README and testing README. However, I am a first time contributor so any extra review would be much appreciated! 

### Testing

I have added a test case to onnx-tests and confirmed that the results from pytorch match the results of burn-import. To ensure the results matched I had to set the random seed for pytorch. I noticed this was missing from the Conv2d python test script which caused to onnx test to fail so I added the forced seed and updated the expected sum in Conv2d. 

Thanks in advance! 
Mitchell

### CODE REVIEW

**Comment by antimora** [comment]
Thank you, @mrvollger, for adding this implementation. I will definitely review it and get back to you. 

So you know, I have been fixing issues with burn-import. I have submitted a couple PRs (pending) and one unsubmitted yet (regarding Reshape). Depending how it goes you may have to merge these changes or we can your changes first, and I'll have to remerge mine. 

I am happy to see more contributors. I have been fixing some bugs when I was importing squeezenet.onnx. Happy to report, I am generating Burn model. As soon as my PRs are merged, I'll submit the model to https://github.com/burn-rs/models.

File: burn-import/src/onnx/op_configuration.rs [hunk 1]
```diff
@@ -28,6 +28,48 @@ pub fn attr_value_f32(value: &AttributeValue, target: &mut f32) {
     }
 }
 
+/// Create a Conv1dConfig from the attributes of the node
+/// TODO check this function carefully, I don't understand it well enough
```
Comment (antimora) [inline burn-import/src/onnx/op_configuration.rs:32]: We need to remove this TODO before merging.

File: burn-import/src/onnx/op_configuration.rs [hunk 2]
```diff
@@ -28,6 +28,48 @@ pub fn attr_value_f32(value: &AttributeValue, target: &mut f32) {
...
+    let StateType::Tensor(tensor) = curr.states.get(0).unwrap().clone().ty;
+
+    // check if the bias is present
+    let bias = curr.states.len() == 2;
+
+    // the channels are inverted in the weight tensor
+    let shape = tensor.shape.unwrap();
+    let channels_in = shape[1];
+    let channels_out = shape[0];
+    //let channels: [usize; 2] = [shape[1], shape[0]];
```
Comment (antimora) [inline burn-import/src/onnx/op_configuration.rs:50]: Dead code should be removed.

File: burn-import/src/onnx/op_configuration.rs [hunk 3]
```diff
@@ -357,3 +399,41 @@ fn padding_config(pads: &[i64]) -> PaddingConfig2d {
...
+
+    if left < 0 || right < 0 {
+        panic!("Negative pad values are not supported");
+    } else if left != right {
+        panic!("Asymmetric padding is not supported");
+    } else if left == right && right == 0 {
+        // i.e [0, 0]
+        PaddingConfig1d::Valid
+    } else if left == right {
+        // i.e [2, 3, 2, 3]
```
Comment (antimora) [inline burn-import/src/onnx/op_configuration.rs:433]: ` // i.e [2, 3, 2, 3]` should be ` // i.e [2, 2] to be accurate.

File: burn-import/src/onnx/op_configuration.rs [hunk 4]
```diff
@@ -357,3 +399,41 @@ fn padding_config(pads: &[i64]) -> PaddingConfig2d {
...
+        PaddingConfig1d::Explicit(left as usize) //#(left as usize, top as usize)
```
Comment (antimora) [inline burn-import/src/onnx/op_configuration.rs:434]: dead code `//#(left as usize, top as usize)` should be removed

File: burn-import/src/onnx/to_burn.rs [hunk 5]
```diff
@@ -453,6 +455,23 @@ impl ONNXGraph {
         )
     }
 
+    /// TODO check that this is valid, I am not 100% sure on it.
```
Comment (antimora) [inline burn-import/src/onnx/to_burn.rs:458]: This todo comment should be removed.

File: burn-import/src/onnx/dim_inference.rs [hunk 6]
```diff
@@ -297,6 +298,23 @@ fn flatten_update_outputs(node: &mut Node) {
     });
 }
 
+/// Infers the shape of a Conv1d node and replaces the shape of the output tensor.
+///
+/// The shape of the output tensor is calculated by running the actual convolution operation.
```
Comment (antimora) [inline burn-import/src/onnx/dim_inference.rs:303]: The comment is outdated and applicable anymore. If you see it on Conv2d, please remove it as well.

Before a big refactor, we used to calculate the output shape by calling the Conv2d code. But we have figured out we do not need to know the shapes. But instead only a dimension of the output is needed. So this comment was forgotten to be cleaned (and probably confusing others now =D)

**Review by antimora — CHANGES_REQUESTED** [review]
You have everything got it correctly. And I am confident because your tests. The onnx file conversation and running guarantees the code's syntax correctness and the actual execution is correct. Even weights and biases are loaded correctly. 

Thanks for adding the seed. It's correct to add it since the weights and biases are initialized from a random data.

**Comment by antimora** [comment]
If we could get the comments fixed, we can merge the PR before mine. I'll take care of the conflicts.

**Comment by mrvollger** [comment]
Awesome, I will get on these requests!

File: burn-import/src/onnx/dim_inference.rs (hunk 6 above)
Comment (mrvollger) [reply burn-import/src/onnx/dim_inference.rs]: Done, I removed it on both functions!

**Comment by mrvollger** [comment]
I am very confused why the tests now fail... They run locally and I only made comment/spacing changes in addressing the review. I will investigate a bit more.

**Comment by antimora** [comment]
> I am very confused why the tests now fail... They run locally and I only made comment/spacing changes in addressing the review. I will investigate a bit more.

The CI is failing because of something else. The clippy got upgraded. We will fix and rerun the tests.

**Review by antimora — APPROVED** [review]
Looks good! Thank for you contribution!

Ready to merge.

**Pull request merged**